crates/subspace-farmer/src/bin/subspace-farmer/commands/farm/dsn.rs

```diff
@@ -192,7 +192,12 @@ pub(super) async fn configure_dsn(
                 .map(|root_blocks| {
                     root_blocks
                         .iter()
-                        .filter_map(|rb| *rb)
+                        .filter_map(|rb| {
+                            if rb.is_none() {
+                                warn!("Received empty optional root block!");
```

Comment:
This is not a warning, this is a hard error, peer will be banned if they reply with this!